GenerateAspireManifest target with value in AspireManifestPublishOutputPath errors out

When setting an `AspireManifestPublishOutputPath` property on the app host and running `dotnet msbuild /t:GenerateAspireManifest` the target fails

```
D:\AppCacheFiles\nuget\aspire.hosting\8.0.0-preview.1.23557.2\build\Aspire.Hosting.targets(118,5): error MSB3073: The c
ommand "..." exited with code -532462766. [D:\source\repos\AspireApp7\AspireApp7.AppHost\AspireApp7.AppHost.csproj]
```

--- a/src/Aspire.Hosting/build/Aspire.Hosting.targets
+++ b/src/Aspire.Hosting/build/Aspire.Hosting.targets
@@ -224,15 +224,24 @@ internal static class ]]>%(ClassName)<![CDATA[
 
   <PropertyGroup>
     <AspirePublisher Condition="'$(AspirePublisher)' == ''">manifest</AspirePublisher>
-    <AspireManifestPublishOutputPath Condition="'$(AspireManifestPublishOutputPath)' == ''">$(_AspireIntermediatePath)manifest.json</AspireManifestPublishOutputPath>
+    <AspireManifestPublishOutputPath Condition="'$(AspireManifestPublishOutputPath)' == ''">$(_AspireIntermediatePath)</AspireManifestPublishOutputPath>
   </PropertyGroup>
 
+  <Target Name="_EnsureAspireManifestPublishOutputPathExists" BeforeTargets="GenerateAspireManifest">
+    <PropertyGroup>
+      <AspireManifestPublishOutputPath>$([MSBuild]::NormalizeDirectory($(AspireManifestPublishOutputPath)))</AspireManifestPublishOutputPath>
+      <_AspireManifestPublishOutputFile>$([MSBuild]::NormalizePath('$(AspireManifestPublishOutputPath)', 'manifest.json'))</_AspireManifestPublishOutputFile>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(AspireManifestPublishOutputPath)" />
+  </Target>
+
   <!-- Entrypoint to allow for generation of an aspire manifest to a given location -->
-  <Target Name="GenerateAspireManifest" DependsOnTargets="Build" Inputs="$(TargetFileName)" Outputs="$(AspireManifestPublishOutputPath)">
+  <Target Name="GenerateAspireManifest" DependsOnTargets="Build;_EnsureAspireManifestPublishOutputPathExists" Inputs="$(TargetFileName)" Outputs="$(_AspireManifestPublishOutputFile)">
     <ItemGroup>
       <_AspireManifestPublishArg Include="%22$(DOTNET_HOST_PATH)%22; %22$(TargetPath)%22;"/>
       <_AspireManifestPublishArg Include="--publisher; $(AspirePublisher);" />
-      <_AspireManifestPublishArg Include="--output-path; %22$(AspireManifestPublishOutputPath)%22" />
+      <_AspireManifestPublishArg Include="--output-path; %22$(_AspireManifestPublishOutputFile)%22" />
     </ItemGroup>
 
     <Exec Command="@(_AspireManifestPublishArg, ' ')"

--- a/src/Aspire.Hosting/build/Aspire.Hosting.targets
+++ b/src/Aspire.Hosting/build/Aspire.Hosting.targets
@@ -227,7 +227,7 @@ internal static class ]]>%(ClassName)<![CDATA[
     <AspireManifestPublishOutputPath Condition="'$(AspireManifestPublishOutputPath)' == ''">$(_AspireIntermediatePath)</AspireManifestPublishOutputPath>
   </PropertyGroup>
 
-  <Target Name="_EnsureAspireManifestPublishOutputPathExists" BeforeTargets="GenerateAspireManifest">
+  <Target Name="_EnsureAspireManifestPublishOutputPathExists">
     <PropertyGroup>
       <AspireManifestPublishOutputPath>$([MSBuild]::NormalizeDirectory($(AspireManifestPublishOutputPath)))</AspireManifestPublishOutputPath>
       <_AspireManifestPublishOutputFile>$([MSBuild]::NormalizePath('$(AspireManifestPublishOutputPath)', 'manifest.json'))</_AspireManifestPublishOutputFile>

